core/query/mixingQuery.mli

```diff
@@ -0,0 +1,93 @@
+open Utility
```

Comment:
Looking at this vs. query.* it seems like there is a lot of overlap/code duplication.  This in turn seems to offer lots of opportunities for regression/having to fix the same bug in two places.

Would it be possible to unify the common parts of query.* with mixingQuery.*, for example by providing two Eval modules, one that handles queries containing Dedup and one without?

Reply:
After a couple attempts, I have decided to move the common parts to a separate module called `QueryLang`. There's still some overlapping, but I can't do better than this while at the same time guaranteeing that the standard normalizer has the same semantics as that in the previous releases of Links.